cdap-master/src/main/java/co/cask/cdap/data/tools/DatasetUpgrader.java

```diff
@@ -96,6 +100,31 @@ public void upgrade() throws Exception {
 
     // Upgrade the datasets instance meta table
     datasetInstanceMDSUpgrader.upgrade();
+
+    // upgrade all the filesets base paths
+    for (DatasetSpecification fileSetSpec : datasetInstanceMDSUpgrader.getFileSetsSpecs()) {
+      upgradeFileSet(fileSetSpec);
+    }
+
+  }
+
+  /**
+   * Upgrades the {@link FileSet} and also its embedded filesets if any by moving the base path under
+   * namespaced directory
+   *
+   * @param dsSpec the {@link DatasetSpecification} of the {@link FileSet} to be upgraded
+   * @throws IOException
+   */
+  private void upgradeFileSet(DatasetSpecification dsSpec) throws IOException {
+    for (DatasetSpecification embeddedDsSpec : dsSpec.getSpecifications().values()) {
+      upgradeFileSet(embeddedDsSpec);
+    }
+    String basePath = FileSetProperties.getBasePath(dsSpec.getProperties());
+    Location oldLocation = locationFactory.create(basePath);
+    Location newlocation = locationFactory.create(Constants.DEFAULT_NAMESPACE)
+      .append(cConf.get(Constants.Dataset.DATA_DIR)).append(basePath);
+    LOG.info("Upgrading base path for dataset {} from {} to {}", dsSpec.getName(), oldLocation, newlocation);
+    renameLocation(oldLocation, newlocation);
```

Comment:
Since I moved this class that method was not needed anymore. Its in upgrade tool and it can use the common shared method which I have already changed according to your suggestions. 
Here is the PR which got merged: https://github.com/caskdata/cdap/pull/1580